sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs

```diff
@@ -1334,5 +1335,32 @@ public virtual async Task<Response<ReleaseKeyResult>> ReleaseKeyAsync(string nam
                 throw;
             }
         }
+
+        /// <summary>
+        /// Get a <see cref="CryptographyClient"/> for the given key.
+        /// </summary>
+        /// <param name="name">The name of the key used to perform cryptographic operations.</param>
+        /// <param name="version">Optional version of the key used to perform cryptographic operations.</param>
+        /// <returns>A <see cref="CryptographyClient"/> using the same options and pipeline as this <see cref="KeyClient"/>.</returns>
+        /// <exception cref="ArgumentException"><paramref name="name"/> is an empty string.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="name"/> is null.</exception>
+        public virtual CryptographyClient GetCryptographyClient(string name, string version = null)
+        {
+            Argument.AssertNotNullOrEmpty(name, nameof(name));
+
+            Uri keyUri = CreateKeyUri(VaultUri, name, version);
+            return new(keyUri, _pipeline);
+        }
+
+        /// <summary>
+        /// Constructs a key <see cref="Uri"/>.
+        /// </summary>
+        /// <param name="vaultUri">The <see cref="Uri"/> to the Key Vault or Managed HSM.</param>
+        /// <param name="name">Name of the key.</param>
+        /// <param name="version">Optional version of the key.</param>
+        /// <returns>A key <see cref="Uri"/>.</returns>
+        internal static Uri CreateKeyUri(Uri vaultUri, string name, string version) => version is null
+            ? new Uri(vaultUri, KeysPath + name)
+            : new Uri(vaultUri, $"{KeysPath}{name}/{version}");
```

Comment:
Can we avoid adding this method by calling `keyClient.GetKey(name, version)` and then using the generated key to create the crypto client?  (That's how I was thinking to implement this for C++) .

Reply:
Not sure about dotnet or C++, but just chatted with JS architects and in JS we would not have an async `get<Subclient>` method and it would need to be async if we want to call keyClient.GetKey

Reply:
Sometimes I forget that C++ is the only lang with all APIs being Sync.  So there's not such restriction for C++ to consider the Async world.

Reply:
The method you highlighted just creates a URI, which your `CryptographyClient` should already take in its public constructor. We also already have methods that get you a `KeyVaultKey`, from which you can get a `JsonWebKey`, which users should be able to pass to a `CryptographyClient` already.

There's no reason to have an async version of any methods I added because methods are sync or async, not classes. That may not be true of Python and Java, however, which have separate classes but might choose to have 1 method on each that returns the same "syncedness" client (that's what I'd do, but talk with your architects).